crates/sp-domains/src/lib.rs

```diff
@@ -246,6 +257,52 @@ impl From<DomainId> for ChainId {
     }
 }
 
+#[derive(Clone, Debug, Decode, Default, Encode, Eq, PartialEq, TypeInfo)]
+pub struct BlockFees<Balance> {
+    /// The consensus chain storage fee
+    pub consensus_storage_fee: Balance,
+    /// The domain execution fee including the storage and compute fee on domain chain,
+    /// tip, and the XDM reward.
+    pub domain_execution_fee: Balance,
+    /// Burned balances on domain chain
+    pub burned_balance: Balance,
+}
+
+impl<Balance> BlockFees<Balance>
+where
+    Balance: CheckedAdd,
+{
+    pub fn new(
+        domain_execution_fee: Balance,
+        consensus_storage_fee: Balance,
+        burned_balance: Balance,
+    ) -> Self {
+        BlockFees {
+            consensus_storage_fee,
+            domain_execution_fee,
+            burned_balance,
+        }
+    }
+
+    /// Returns the total fees that was collected and burned on the Domain.
+    pub fn total_fees(&self) -> Option<Balance> {
+        self.consensus_storage_fee
+            .checked_add(&self.domain_execution_fee)
+            .and_then(|balance| balance.checked_add(&self.burned_balance))
+    }
+}
+
+/// Type that holds the transfers(in/out) for a given chain.
+#[derive(Debug, Decode, Encode, TypeInfo, PartialEq, Eq, Clone, Default)]
+pub struct Transfers<Balance> {
```

Comment:
Perhaps we also need a limit to the size of the following `BTreeMap` since they are also store on state and need to handle one by one upon ER confirmation.